Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp

```diff
@@ -952,7 +958,21 @@ void View::setCursor(const WebCore::Cursor& cursor)
     WebCore::IntPoint hotspot = WebCore::determineHotSpot(image.get(), cursor.hotSpot());
     wpe_view_set_cursor_from_bytes(m_wpeView.get(), bytes.get(), width, height, stride, hotspot.x(), hotspot.y());
 #elif USE(SKIA)
-    // FIXME: implement.
+    auto nativeImage = cursor.image()->nativeImageForCurrentFrame();
+    if (!nativeImage)
+        return;
+
+    SkPixmap pixmap;
+    auto platformImage = nativeImage->platformImage();
+    ASSERT(platformImage->peekPixels(&pixmap));
+
+    GRefPtr<GBytes> bytes = adoptGRef(g_bytes_new_with_free_func(pixmap.addr(), pixmap.computeByteSize(), [](gpointer data) {
+        static_cast<SkImage*>(data)->unref();
+    }, platformImage.get()));
+    platformImage->ref();
```

Comment:
Same here, I would call ref before. It's not very important and probably just a matter of taste, but visually you see the ref first and then the unref. In any case, don't change it if you like this way more.

Reply:
Another option here is calling `SkRef` that calls `ref()` and returns the pointer so you can pass it directly to `g_bytes_new_with_free_func`